test/distributed/test_symmetric_memory.py

```diff
@@ -90,6 +90,19 @@ def _verify_symmetric_memory(self, symm_mem):
 
         symm_mem.barrier()
 
+    @skipIfRocm
+    @skip_if_lt_x_gpu(2)
+    def test_cuda_nvlink_connectivity_detection(self) -> None:
+        from torch._C._autograd import DeviceType
+        from torch._C._distributed_c10d import _detect_dma_connectivity
+
+        connectivity = _detect_dma_connectivity(DeviceType.CUDA, "nvlink")
+        self.assertEqual(connectivity.device_type, DeviceType.CUDA)
+        self.assertEqual(connectivity.connection_type, "nvlink")
+        self.assertEqual(len(connectivity.matrix), torch.cuda.device_count())
+        for row in connectivity.matrix:
+            self.assertEqual(len(row), torch.cuda.device_count())
```

Comment:
For 8 GPUs, if `self.world_size == 2`, would `len(row)` change at all?

Reply:
are we not checking value == 18 because we do not know what hardware it is ?

Reply:
> For 8 GPUs, if self.world_size == 2, would len(row) change at all?

Right. This is checking the connectivity among visible cuda devices. For checking the connectivity among ranks, we need to build another layer on top.

> are we not checking value == 18 because we do not know what hardware it is ?

Yes that's right.